### DISCUSSION TITLE
node_modules persist but ruby gems do not

### DISCUSSION THREAD

**Discussion by markgoho** [community/community#17329] [Codespaces]
After the very first time building a container and running `npm install` it seems the `node_modules` folder persists. I believe this is because the entire root directory is being persisted.

How do I do the same thing for ruby gems?

**Comment by legomushroom** [comment]
@markgoho mind elaborating on your dev flow, what do you mean by `node_modules` folder persist(or how do you check that)? 

Note that we run `npm install` on Codespace creation automatically so it might seem like the folder persists while it is not 🤔

**Reply by markgoho** [reply]
for example, when I `rebuild container` via the command palette, the `workspace/node_modules` folder is still there, but `/usr/local/bundle` is removed and I have to run `bundle install` again

if you automatically run `npm install` that would make sense why the node_modules folder would still be there, but I assumed it was because you just made sure everything in the `/workspace` root was saved between container rebuilds

UPDATE: when I `rm -rf node_modules` and then rebuild the container, the folder is _not present_ but then, when I run `yarn` and then rebuild the container, the folder remains.
